lib/avo/resources/base.rb

```diff
@@ -462,6 +464,25 @@ def fill_record(record, params, extra_params: [])
         record
       end
 
+      def fill_join_record(record:, fields:, params:, extra_params:)
+        # Write the field values
+        params.each do |key, value|
+          field = fields.find { |f| f.id == key.to_sym }
+
+          next unless field.present?
+
+          record = field.fill_field record, key, value, params
+        end
+
+        # Write the user configured extra params to the record
+        if extra_params.present?
+          # Let Rails fill in the rest of the params
+          record.assign_attributes params.permit(extra_params)
+        end
+
+        record
+      end
```

Comment:
There is only 1 line different from `fill_record`. Can we adapt the `fill_record` method to also work with this? Something like: (UNTESTED)

```ruby
      def fill_record(record, params, extra_params: [], fields: nil)
        # Write the field values
        params.each do |key, value|
          field = if fields.present?
            fields.find { |f| f.id == key.to_sym }
          else
            fields_by_database_id[key]
          end

          next unless field.present?

          record = field.fill_field record, key, value, params
        end

        # Write the user configured extra params to the record
        if extra_params.present?
          # Let Rails fill in the rest of the params
          record.assign_attributes params.permit(extra_params)
        end

        record
      end
```

Otherwise we need to maintain 2 identic methods

Reply:
@Paul-Bob I pushed just now addressing all of your comments. Could you take a look please? :pray: